workstation-backend/component/models.py

```diff
@@ -1,11 +1,12 @@
+from uuid import uuid4
+
 from django.db import models
 from workspace import models as workspace_models
 
 # Create your models here.
 class BaseComponent(models.Model):
-    workspace_id = models.ForeignKey(
-        workspace_models.Workspace, on_delete=models.CASCADE
-    )
+    id = models.UUIDField(primary_key=True, default=uuid4)
+    workspace = models.ForeignKey(workspace_models.Workspace, on_delete=models.CASCADE)
```

Comment:
Shouldn't this be OneToOne since any component can only belong to one workspace?

Reply:
So if this is a OneToOne, the component can only have one workspace AND the workspace can only have one component. What we want is actually: the component can only have one workspace AND the workspace can have multiple components.

https://stackoverflow.com/questions/13115207/differences-between-one-to-one-and-foreign-key-relationships